### PR TITLE
Migrate Options to Kotlin

### PR DESCRIPTION
**Pull request by swankjesse**
Also switch japicmp to compare against Okio 1.14.1, which adds a bunch of redundant
final modifiers for its benefit

### CODE REVIEW

File: okio/jvm/src/main/java/okio/Options.kt [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okio
+
+import java.util.AbstractList
```
Comment (Egorand) [inline okio/jvm/src/main/java/okio/Options.kt]: Both of these are available in Kotlin as typealiases

File: okio/jvm/src/test/java/okio/BufferedSourceTest.java [hunk 2]
```diff
@@ -882,7 +882,7 @@ private void assertLongDecimalString(String s, long expected) throws IOException
   }
 
   @Test public void select() throws IOException {
-    Options options = Options.of(
+    Options options = Options.Companion.of(
```
Comment (Egorand) [inline okio/jvm/src/test/java/okio/BufferedSourceTest.java]: `Companion` is redundant (this was probably auto-generated, but still)

**Comment by Egorand** [comment]
Javadoc task fails - it can't find Java files! 🎆

File: okio/jvm/src/main/java/okio/Options.kt (hunk 1 above)
Comment (swankjesse) [reply okio/jvm/src/main/java/okio/Options.kt]: Will fix in follow-up

File: okio/jvm/src/test/java/okio/BufferedSourceTest.java (hunk 2 above)
Comment (swankjesse) [reply okio/jvm/src/test/java/okio/BufferedSourceTest.java]: Will fix in follow-up.

**Pull request merged**